### ISSUE TITLE
Errors installing on Windows 

### ISSUE THREAD

**Issue by Mnzs1701** [facebookresearch/DRTK#6]
Thank you for your contribution, I was hoping to use DRTK in our visualization pipeline and we need to run it on a Windows machine (for access to Quest Link). However we are facing some installation errors

```
nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Fri_Jun_14_16:44:19_Pacific_Daylight_Time_2024
Cuda compilation tools, release 12.6, V12.6.20
Build cuda_12.6.r12.6/compiler.34431801_0
```
Torch is able to find cuda
```
>>> import torch
>>> torch.__version__
'2.4.1'
>>> torch.cuda.is_available()
True
```
Running `pip install .` fails with many errors including:
```
      C:\username\miniconda\envs\fall\Library\include\cuda_fp16.hpp(1893): error: asm operand type size(8) does not match type/size implied by constraint 'r'
            asm ("ld.global.nc.b32 %0, [%1];"  : "=r"(*(reinterpret_cast<unsigned int *>(&(ret)))) : "r"(ptr));
                                                                                                     ^

      C:\username\miniconda\envs\fall\Library\include\cuda_fp16.hpp(1899): error: asm operand type size(8) does not match type/size implied by constraint 'r'
            asm ("ld.global.nc.b16 %0, [%1];"  : "=h"(*(reinterpret_cast<unsigned short *>(&(ret)))) : "r"(ptr));
                                                                                                       ^

      C:\username\miniconda\envs\fall\Library\include\cuda_fp16.hpp(1905): error: asm operand type size(8) does not match type/size implied by constraint 'r'
            asm ("ld.global.cg.b32 %0, [%1];"  : "=r"(*(reinterpret_cast<unsigned int *>(&(ret)))) : "r"(ptr));
                                                                                                     ^

      C:\username\miniconda\envs\fall\Library\include\cuda_fp16.hpp(1911): error: asm operand type size(8) does not match type/size implied by constraint 'r'
            asm ("ld.global.cg.b16 %0, [%1];"  : "=h"(*(reinterpret_cast<unsigned short *>(&(ret)))) : "r"(ptr));
```

There are more errors and many warnings
```
 C:/username/miniconda/envs/fall/Lib/site-packages/torch/include\c10/core/TensorImpl.h(3155): error: static assertion failed with "Actual and Expected sizes of a field did not match!"
            static_assert(
            ^
                detected during:
                  instantiation of "bool c10::C10_TensorImpl_Size_Check_Dummy_Class<cplusplus, clang_ver_major, gcc_ver, gcc_ver_minor, nvcc, cuda_version, cuda_version_major, ptr_size>::are_equal<Actual,Expected,FiledName>() [with cplusplus=199711U, clang_ver_major=0U, gcc_ver=0U, gcc_ver_minor=0U, nvcc=1U, cuda_version=12060U, cuda_version_major=0U, ptr_size=8U, Actual=8U, Expected=4U, FiledName=c10::C10_TensorImpl_Size_Check_Dummy_Class<199711U, 0U, 0U, 0U, 1U, 12060U, 0U, 8U>::FieldNameEnum::extra_meta_]" at line 3195
                  instantiation of "bool c10::C10_TensorImpl_Size_Check_Dummy_Class<cplusplus, clang_ver_major, gcc_ver, gcc_ver_minor, nvcc, cuda_version, cuda_version_major, ptr_size>::check_sizes() [with cplusplus=199711U, clang_ver_major=0U, gcc_ver=0U, gcc_ver_minor=0U, nvcc=1U, cuda_version=12060U, cuda_version_major=0U, ptr_size=8U]" at line 3244
```

and

```
     C:/username/miniconda/envs/fall/Lib/site-packages/torch/include\ATen/core/boxing/impl/boxing.h(177): error: type name is not allowed
            return std::make_tuple((std::move(stack[indices]).to<Types>())...);
                                                                 ^

      C:/username/miniconda/envs/fall/Lib/site-packages/torch/include\ATen/core/boxing/impl/boxing.h(177): error: expected an expression
            return std::make_tuple((std::move(stack[indices]).to<Types>())...);
```

**Comment by podgorskiy** [comment]
Are you compiling this with MSVC? It seems like either a version mismatch or MSVC is misconfigured in our setup.py.  All mentioned errors are within CUDA headers and Torch headers, nothing specific to DRTK.

What compiler are you using? Did you make any changes to setup.py?

**Comment by Mnzs1701** [comment]
I have the following MSVC configuration
```
>cl.exe /?
Microsoft (R) C/C++ Optimizing Compiler Version 19.41.34120 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```
I managed to get DRTK installed on my system albeit with some workarounds, I am not sure if this is something needed on all systems.

Some of the changes I needed to make were
1. I had to change the Windows Compile Flags to `"/MD", "/GR",` from `"/MT", "/GR-",` in `setup.py`
2. I had to update to the nightly version of pytorch, nvcc complains that there are multiple std flags set and uses the last one. Which pytorch sets to stdc++17 as given [here](https://github.com/pytorch/pytorch/blob/c0e98a485ba32e95259860aabc379898aa141ff2/torch/utils/cpp_extension.py#L2240C1-L2240C83) this conflicts with the stdc++20 used in setup.py 
3. There were some other changes that I made, with a full guide [here](https://gist.github.com/Mnzs1701/c0a150b3a183b91adb844bada21cc2c6) these could be specific to my system but I will leave it here just in case someone else has the same issues.
